### PR TITLE
Add ability to read weights from a gz file.

### PR DESCRIPTION
**Pull request by Tilps**

### CODE REVIEW

File: src/Network.cpp [hunk 1]
```diff
@@ -325,25 +342,41 @@ std::pair<int, int> Network::load_network_file(std::string filename) {
...
+    // Check for gzip id bytes. 31, 139.
+    if (wtfile.get() != 31) {
+        wtfile.unget();
+    } else if (wtfile.get() != 139) {
+        wtfile.unget();
+        wtfile.unget();
+    } else {
+        // File is gzipped, reopen with zlib.
+        wtfile.close();
+        auto gzhandle = gzopen(filename.c_str(), "rb");
```
Comment (glinscott) [inline src/Network.cpp]: I think gzopen works for uncompressed files too?  That way we can simplify this codepath and not have to check for gz explicitly.

Comment (Tilps) [reply src/Network.cpp]: Yes that is apparently true.  I'll make an update later.

**Comment by Tilps** [comment]
Updated for Gary's feedback.

**Comment by glinscott** [comment]
Thanks!

**Pull request merged**